File: src/main/java/eu/qwsome/sql/condition/UnaryCondition.java
package eu.qwsome.sql.condition;

/**
 * @author Lukáš Kvídera
 */
abstract class UnaryCondition implements Condition {

  private final ValueHolder field;

  UnaryCondition(final ValueHolder column) {
    this.field = column;
  }

  @Override
  public ValueConstructor getValues() {
    return new ValueConstructor().add(this.field);
  }

  @Override
  public void appendTo(final StringBuilder builder) {
    builder.append(this.field.getSql())
      .append(getSuffix());
  }

  abstract String getSuffix();
}

File: src/main/kotlin/eu/qwsome/sql/condition/UnaryCondition.kt
/*
 * @author Lukáš Kvídera
 */
package eu.qwsome.sql.condition

abstract class UnaryCondition(private val field: ValueHolder) : Condition {

    override fun getValues(): ValueConstructor {
        return ValueConstructor().add(field)
    }

    override fun appendTo(builder: StringBuilder) {
        builder.append(field.getSql())
            .append(getSuffix())
    }

    protected abstract fun getSuffix(): String
}